x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go

```diff
@@ -219,7 +257,7 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 }
 
 // createAwsRequiredClients will return the two necessary client instances to do Metric requests to the AWS API
-func (m *MetricSet) createAwsRequiredClients(beatsConfig awssdk.Config, regionName string, config aws.Config) (*cloudwatch.Client, *resourcegroupstaggingapi.Client, error) {
+func (m *MetricSet) createAwsRequiredClients(beatsConfig awssdk.Config, regionName string, config aws.Config) (*cloudwatch.Client, *resourcegroupstaggingapi.Client, *apigateway.Client, *apigatewayv2.Client, error) {
```

Comment:
nit - should we create a return struct here instead of adding more return values ?

Reply:
Changed it to struct in  https://github.com/orgs/elastic/projects/1228/views/18
I liked the idea. I might need to test some other namespace maybe to see if we break sth? Integration tests pass

Reply:
Nice :) . `createAwsRequiredClients` is not exported so shouldn't cause any issues :)